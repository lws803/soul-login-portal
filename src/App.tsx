import React from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { ChakraProvider } from '@chakra-ui/react';

import Login from './Pages/Login';
import Register from './Pages/Register';
import EmailConfirmation from './Pages/EmailConfirmation';
import ResetPassword from './Pages/ResetPassword';
import RequestPasswordReset from './Pages/RequestPasswordReset';
import RequestEmailVerification from './Pages/RequestEmailVerification';
import theme from './theme';

function App() {
  return (
    <ChakraProvider theme={theme}>
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Login />} />
          <Route path="/register" element={<Register />} />
          <Route path="/verify-email" element={<EmailConfirmation />} />
          <Route path="/reset-password" element={<ResetPassword />} />
          <Route
            path="/request-password-reset"
            element={<RequestPasswordReset />}
          />
          <Route
            path="/request-email-verification"
            element={<RequestEmailVerification />}
          />
        </Routes>
      </BrowserRouter>
    </ChakraProvider>
  );
}

export default App;
