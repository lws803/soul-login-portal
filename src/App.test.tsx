import React from 'react';
import { render } from '@testing-library/react';

import App from './App';

test('renders login page', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
