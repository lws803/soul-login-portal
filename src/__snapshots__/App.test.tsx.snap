// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login page 1`] = `
<div>
  <div
    class="css-gmuwbf"
  >
    <div
      class="css-s8l1zf"
    >
      <form>
        <div
          class="chakra-form-control css-1kxonj9"
          role="group"
        >
          <label
            class="chakra-form__label css-2gx1h6"
            for="email"
            id="field-1-label"
          >
            Email
          </label>
          <input
            aria-describedby="field-1-helptext"
            class="chakra-input css-1c6j008"
            disabled=""
            id="email"
            name="email"
            value=""
          />
          <div
            class="chakra-form__helper-text css-1y1zyf5"
            id="field-1-helptext"
          >
            We'll never share your email.
          </div>
        </div>
        <div
          class="chakra-form-control css-kacrke"
          role="group"
        >
          <label
            class="chakra-form__label css-2gx1h6"
            for="password"
            id="field-2-label"
          >
            Password
          </label>
          <input
            class="chakra-input css-1c6j008"
            disabled=""
            id="password"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="chakra-stack css-1a5csaf"
        >
          <button
            class="chakra-button css-f6n2t5"
            disabled=""
            type="submit"
          >
            Login
          </button>
          <button
            class="chakra-button css-f6n2t5"
            disabled=""
            type="button"
          >
            Register
          </button>
        </div>
      </form>
      <div
        class="chakra-stack css-1iiq7kh"
      >
        <a
          class="chakra-link css-f4h6uy"
          href="/request-password-reset"
        >
          Request password reset
        </a>
        <a
          class="chakra-link css-f4h6uy"
          href="/request-email-verification"
        >
          Request email verification
        </a>
      </div>
      <p
        class="chakra-text css-4j7iwf"
      >
        PlatformId and callback is not present.
      </p>
    </div>
  </div>
</div>
`;
